### PR TITLE
Added support for kfs rest infer endpoint

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

File: src/http_rest_api_handler.cpp [hunk 1]
```diff
@@ -131,8 +138,93 @@ void HttpRestApiHandler::registerAll() {
...
+                Value value(i->value.GetBool());
+                param.AddMember("bool_param", value, doc.GetAllocator());
+                scope["parameters"].RemoveMember(i->name.GetString());
+                scope["parameters"].AddMember(name, param, doc.GetAllocator());
+            }
+        }
+    }
+}
+
+Status HttpRestApiHandler::processInferKFSRequest(const HttpRequestComponents& request_components, std::string& response, const std::string& response_body) {
```
Comment (michalkulakowski) [inline src/http_rest_api_handler.cpp:175]: request_body?

**Review by atobiszei — COMMENTED** [review]
Add either test for simple inference or test veryfing correct translation of rest message into grpc infer request.

File: src/http_rest_api_handler.cpp [hunk 2]
```diff
@@ -131,9 +138,104 @@ void HttpRestApiHandler::registerAll() {
...
+    return grpc_request;
+}
+
+Status HttpRestApiHandler::processInferKFSRequest(const HttpRequestComponents& request_components, std::string& response, const std::string& request_body) {
+    std::string modelName(request_components.model_name);
+    std::string modelVersion(std::to_string(request_components.model_version.value_or(0)));
+    SPDLOG_DEBUG("Processing REST request for model: {}; version: {}", modelName, modelVersion);
+    ::inference::ModelInferRequest grpc_request(prepareGrpcRequest(modelName, modelVersion, request_body));
+    ::inference::ModelInferResponse grpc_response;
+    const ::grpc::Status gstatus = kfsGrpcImpl.ModelInfer(nullptr, &grpc_request, &grpc_response);
```
Comment (atobiszei) [inline src/http_rest_api_handler.cpp:226]: We cannot always respond with the same status code when inference does not pass. Probably we need to:
1) change grpc prediction service to have function that returns ovms::Status, and second one calling it transforming ovms::Status -> grpc::Status.

This comment applies as well to ModelReady.

**Review by michalkulakowski — APPROVED** [review]

File: src/http_rest_api_handler.cpp (hunk 2 above)
Comment (atobiszei) [reply src/http_rest_api_handler.cpp:226]: This can be fixed in further MR.

**Review by atobiszei — APPROVED** [review]

**Pull request merged**